internal/db/dqlite.go

```diff
@@ -349,7 +366,10 @@ func dqliteNetworkDial(ctx context.Context, addr string, db *DB) (net.Conn, erro
 
 // Stop closes the database and dqlite connection.
 func (db *DB) Stop() error {
+	db.statusLock.Lock()
 	db.cancel()
+	db.status = StatusOffline
```

Comment:
After stopping the database I think it's confusing if you get back the error `"Database is not yet initialized"`. This indicates you could still recover from the stop. 
Do we use it like this or is a call to `Stop()` meant as a final operation as part of shutting down the daemon?
If so would it make more sense to have a `StatusShutdown` instead?

Reply:
Sure, I can add a `StatusShutdown` here.

Reply:
Instead, I've made `StatusOffline` report `Database is offline` and added a `StatusNotReady` to indicate the database is not yet initialized.